File: api/src/main/java/no/difi/meldingsutveksling/api/ConversationStrategy.java
package no.difi.meldingsutveksling.api;

import no.difi.meldingsutveksling.nextmove.NextMoveException;
import no.difi.meldingsutveksling.nextmove.NextMoveOutMessage;

public interface ConversationStrategy {

    void send(NextMoveOutMessage message) throws NextMoveException;

}

File: api/src/main/kotlin/no/difi/meldingsutveksling/api/ConversationStrategy.kt
package no.difi.meldingsutveksling.api

import no.difi.meldingsutveksling.nextmove.NextMoveException
import no.difi.meldingsutveksling.nextmove.NextMoveOutMessage

interface ConversationStrategy {
    @Throws(NextMoveException::class)
    fun send(message: NextMoveOutMessage)
}

interface DpvConversationStrategy: ConversationStrategy {}

interface DpoConversationStrategy: ConversationStrategy {}

interface DpeConversationStrategy: ConversationStrategy {}

interface DpfConversationStrategy: ConversationStrategy {}

interface DpiConversationStrategy: ConversationStrategy {}

